src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Rules/XamlAppDef.xaml

```diff
@@ -0,0 +1,91 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<!--Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.-->
+<Rule
+  Name="XamlAppDef"
+  DisplayName="General"
+  PageTemplate="generic"
+  Description="File Properties"
+  xmlns="http://schemas.microsoft.com/build/2009/properties">
+  <Rule.DataSource>
+    <DataSource Persistence="ProjectFile" HasConfigurationCondition="False" ItemType="XamlAppDef" SourceOfDefaultValue="AfterContext" />
+  </Rule.DataSource>
+  <Rule.Categories>
+    <Category Name="Advanced" DisplayName="Advanced" />
+    <Category Name="Misc" DisplayName="Misc" />
+  </Rule.Categories>
+
+  <DynamicEnumProperty Name="{}{ItemType}" DisplayName="Build Action"  Category="Advanced"
+                     Description="How the file relates to the build and deployment processes."
+                     EnumProvider="ItemTypes" />
+  <EnumProperty
+      Name="CopyToOutputDirectory"
+      DisplayName="Copy to Output Directory"
+      Category="Advanced"
+      Description="Specifies the source file will be copied to the output directory.">
+    <EnumValue Name="Never" DisplayName="Do not copy" />
+    <EnumValue Name="Always" DisplayName="Copy always" />
+    <EnumValue Name="PreserveNewest" DisplayName="Copy if newer" />
+  </EnumProperty>
+
+  <StringProperty
+      Name="Generator"
+      Category="Advanced"
+      DisplayName="Custom Tool"
+      Description="Specifies the tool that transforms a file at design time and places the output of that transformation into another file. For example, a dataset (.xsd) file comes with a default custom tool." />
+  <StringProperty
+      Name="CustomToolNamespace"
+      Category="Advanced"
+      DisplayName="Custom Tool Namespace"
+      Description="The namespace into which the output of the custom tool is placed." />
+
+  <StringProperty
+      Name="Identity"
+      Visible="false"
+      ReadOnly="true"
+      Category="Misc"
+      Description="The item specified in the Include attribute.">
+    <StringProperty.DataSource>
+      <DataSource Persistence="Intrinsic" ItemType="None" PersistedName="Identity" SourceOfDefaultValue="AfterContext" />
```

Comment:
Replace every usage of `ItemType="None"` with `ItemType="XamlAppDef"`.

Reply:
Even better, delete the `ItemType` from all properties. This is automatically pulled from the DataSource on the rule itself.

Reply:
Filed to https://github.com/dotnet/project-system/issues/3815 to basically do this across all the rules so we never run into this again.

Reply:
Also filed https://github.com/dotnet/project-system/issues/3816. We have tracing for a bunch of this, but we hook up too late for it to be output into the Output window.

Reply:
Strike removing the ItemType, I'm replacing it with XamlAppDef. It's required for the `BrowseObject` but not for the `File` context.